Tests/GRPCHTTP2CoreTests/Client/Connection/ClientConnectionHandlerStateMachineTests.swift

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2024, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import NIOCore
+import NIOEmbedded
+import XCTest
+
+@testable import GRPCHTTP2Core
+
+final class ClientConnectionHandlerStateMachineTests: XCTestCase {
+  private func makeStateMachine(
+    keepAliveWithoutCalls: Bool = false
+  ) -> ClientConnectionHandler.StateMachine {
+    return ClientConnectionHandler.StateMachine(allowKeepAliveWithoutCalls: keepAliveWithoutCalls)
+  }
+
+  func testCloseAllStreamsWhenActive() {
+    var state = self.makeStateMachine()
+    state.streamOpened(1)
+    XCTAssertEqual(state.streamClosed(1), .startIdleTimer(cancelKeepAlive: true))
+  }
+
+  func testCloseSomeStreamsWhenActive() {
+    var state = self.makeStateMachine()
+    state.streamOpened(1)
+    state.streamOpened(2)
+    XCTAssertEqual(state.streamClosed(2), .none)
+  }
```

Comment:
Can we merge these two tests like you've done in `testCloseSomeStreamsWhenClosing`? I think asserting that going from 0 -> 1 -> 2  to 2 -> 1 -> 0 streams and asserting what the state at each step should look like is a bit more complete than having two separate tests.